File: Server/src/main/content/global/skill/production/cooking/recipe/stew/StewRecipe.java
package content.global.skill.production.cooking.recipe.stew;

import content.global.skill.production.cooking.recipe.Recipe;
import core.game.interaction.NodeUsageEvent;
import core.game.node.entity.player.Player;
import core.game.node.item.Item;

/**
 * The Stew recipe.
 */
public class StewRecipe extends Recipe {

    private static final Item UNCOOKED_STEW = new Item(2001);

    private static final Item BOWL_OF_WATER = new Item(1921);

    private static final Item MEAT = new Item(2142);

    private static final Item POTATO = new Item(1942);

    private static final Item INCOMPLETE_STEW = new Item(1997);

    private static final Item INCOMPLETE_STEW2 = new Item(1999);

    @Override
    public void mix(Player player, NodeUsageEvent event) {
        Item first = event.getUsedItem();
        Item second = event.getBaseItem();
        if (first != null && second != null) {
            if (player.getInventory().remove(first) && player.getInventory().remove(second)) {
                if (first.getId() == BOWL_OF_WATER.getId() || second.getId() == BOWL_OF_WATER.getId()) {
                    player.getInventory().add(first.getId() == POTATO.getId() ? INCOMPLETE_STEW : first.getId() == MEAT.getId() ? INCOMPLETE_STEW2 : second.getId() == POTATO.getId() ? INCOMPLETE_STEW : second.getId() == MEAT.getId() ? INCOMPLETE_STEW2 : null);
                } else {
                    player.getInventory().add(UNCOOKED_STEW);
                }
                player.getPacketDispatch().sendMessage(getMixMessage(event));
            }
        }
    }

    @Override
    public Item getBase() {
        return BOWL_OF_WATER;
    }

    @Override
    public Item getProduct() {
        return UNCOOKED_STEW;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{MEAT, POTATO, MEAT, POTATO};
    }

    @Override
    public Item[] getParts() {
        return new Item[]{BOWL_OF_WATER, BOWL_OF_WATER, INCOMPLETE_STEW, INCOMPLETE_STEW2};
    }

    @Override
    public String getMixMessage(NodeUsageEvent event) {
        return "You cut up the " + (event.getUsedItem().getName().toLowerCase().contains("incomplete") ? event.getBaseItem().getName().toLowerCase() : event.getUsedItem().getName().toLowerCase().replace("cooked", "").trim()) + " and put it into the stew.";
    }

    @Override
    public boolean isSingular() {
        return false;
    }

}

File: Emulator/src/main/content/global/skill/cooking/type/stew/StewRecipe.kt
package content.global.skill.cooking.type.stew

import content.global.skill.cooking.type.Recipe
import org.rs.consts.Items
import core.api.removeItem
import core.api.sendMessage
import core.game.interaction.NodeUsageEvent
import core.game.node.entity.player.Player
import core.game.node.item.Item

/**
 * Stew recipe.
 */
class StewRecipe : Recipe() {

    override fun mix(player: Player, event: NodeUsageEvent) {
        val first = event.usedItem
        val second = event.baseItem
        if (first != null && second != null) {
            if (removeItem(player, first) && removeItem(player, second)) {
                if (first.id == BOWL_OF_WATER.id || second.id == BOWL_OF_WATER.id) {
                    player.inventory.add(if (first.id == POTATO.id) INCOMPLETE_STEW else if (first.id == COOKED_MEAT.id) INCOMPLETE_STEW2 else if (second.id == POTATO.id) INCOMPLETE_STEW else if (second.id == COOKED_MEAT.id) INCOMPLETE_STEW2 else null)
                } else {
                    player.inventory.add(UNCOOKED_STEW)
                }
                sendMessage(player, getMixMessage(event))
            }
        }
    }

    override fun getBase(): Item {
        return BOWL_OF_WATER
    }

    override fun getProduct(): Item {
        return UNCOOKED_STEW
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(COOKED_MEAT, POTATO, COOKED_MEAT, POTATO)
    }

    override fun getParts(): Array<Item> {
        return arrayOf(BOWL_OF_WATER, BOWL_OF_WATER, INCOMPLETE_STEW, INCOMPLETE_STEW2)
    }

    override fun getMixMessage(event: NodeUsageEvent): String {
        return "You cut up the " + (if (event.usedItem.name.lowercase()
                .contains("incomplete")
        ) event.baseItem.name.lowercase() else event.usedItem.name.lowercase().replace("cooked", "")
            .trim { it <= ' ' }) + " and put it into the stew."
    }

    override fun isSingular(): Boolean {
        return false
    }

    companion object {
        private val UNCOOKED_STEW = Item(Items.UNCOOKED_STEW_2001)
        private val BOWL_OF_WATER = Item(Items.BOWL_OF_WATER_1921)
        private val COOKED_MEAT = Item(Items.COOKED_MEAT_2142)
        private val POTATO = Item(Items.POTATO_1942)
        private val INCOMPLETE_STEW = Item(Items.INCOMPLETE_STEW_1997)
        private val INCOMPLETE_STEW2 = Item(Items.INCOMPLETE_STEW_1999)
    }
}
